packages/web3-eth-contract/types/index.d.ts

```diff
@@ -22,8 +22,8 @@ import {Common, PromiEvent, provider, hardfork, chain, BlockNumber, PastLogsOpti
 import {Accounts} from 'web3-eth-accounts';
 import {AbiItem} from 'web3-utils';
 
-// TODO: Add generic type!
-export class Contract {
+// Here you can provide generated types for methods and events from your contract
+export class Contract<Methods = any, Events = any> {
```

Comment:
I don't think there should be any restrictions. Everyone will generate types with different libraries, methods will get something like this object:

```typescript
 interface ContractTypes {
  /**
   * Payable: true
   * Constant: false
   * StateMutability: payable
   * Type: function
   */
  enter(): MethodPayableReturnContext;
  /**
   * Payable: false
   * Constant: true
   * StateMutability: view
   * Type: function
   */
  getManager(): MethodConstantReturnContext<string>;
  /**
   * Payable: false
   * Constant: true
   * StateMutability: view
   * Type: function
   */
  getPlayers(): MethodConstantReturnContext<string[]>;
  /**
   * Payable: false
   * Constant: false
   * StateMutability: nonpayable
   * Type: function
   */
  pickWinner(): MethodReturnContext;
  /**
   * Payable: false
   * Constant: true
   * StateMutability: view
   * Type: function
   * @param parameter0 Type: uint256, Indexed: false
   */
  players(parameter0: string): MethodConstantReturnContext<string>;
}
```

It will be clear and comfortable, I can add some examples in the code

Reply:
Ok, agree, in a common case it will be`Record<string, unknown>` or do you have another idea how to add typings to it?